cpp-client/deephaven/client/include/private/deephaven/client/server/server.h

```diff
@@ -220,21 +224,29 @@ class Server : public std::enable_shared_from_this<Server> {
   void sendRpc(const TReq &req, std::shared_ptr<SFCallback<TResp>> responseCallback,
       TStub *stub, const TPtrToMember &pm);
 
-  std::pair<std::string, std::string> getAuthHeader() const;
+  template<typename Fun>
+  void forEachHeaderNameAndValue(Fun fun) {
```

Comment:
suggest making this not a template:

```
#include <functional>
```
at top of file and then change this declaration to
```
 void forEachHeaderNameAndValue(const std::function<void(const std::string &, const std::string &)> &fun);
```

and then move the body to server.cc.  Your callers should not need to change.

Reply:
Much better.